### DISCUSSION TITLE
Installation issue using pip

### DISCUSSION THREAD

**Discussion by skoudoro** [mlcommons/GaNDLF#735] [Q&A] [answered]
Hi Gandlf Team,

I tried to install gandlf via `pip install gandlf` and it fails with the following error:

```python
❯ pip install gandlf
Collecting gandlf
  Downloading GANDLF-0.0.1720231112-py3-none-any.whl.metadata (8.7 kB)
INFO: pip is looking at multiple versions of gandlf to determine which version is compatible with other requirements. This could take a while.
  Downloading GANDLF-0.0.17-py3-none-any.whl.metadata (8.7 kB)
  Downloading GANDLF-0.0.16-py3-none-any.whl.metadata (8.6 kB)
ERROR: Cannot install gandlf==0.0.16, gandlf==0.0.17 and gandlf==0.0.1720231112 because these package versions have conflicting dependencies.

The conflict is caused by:
    gandlf 0.0.1720231112 depends on torch==1.13.1
    gandlf 0.0.17 depends on torch==1.13.1
    gandlf 0.0.16 depends on torch==1.13.1

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflicts
```

So I decided restart from a fresh virtual env and clone the repo. I did `pip install .` from the source and it still fails.

```python
❯ pip install .
Processing /Users/skoudoro/devel/GaNDLF
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
INFO: pip is looking at multiple versions of gandlf to determine which version is compatible with other requirements. This could take a while.
ERROR: Could not find a version that satisfies the requirement torch==1.13.1 (from gandlf) (from versions: 2.0.0, 2.0.1, 2.1.0)
ERROR: No matching distribution found for torch==1.13.1
```

I do not plan to use docker and I did not try the conda package. Is there a way to install GaNDLF using pip?

Also, Does it support a recent version of Numpy and pytorch ?

EDIT:  use python 3.11

**Comment by sarthakpati** [comment, marked as answer]
Thanks for your interest in GaNDLF, @skoudoro!

We are currently working on updating the dependencies [[ref](https://github.com/mlcommons/GaNDLF/issues/726)]. In the meantime, I will suggest you follow the installation instructions as given in the guide [[ref](https://mlcommons.github.io/GaNDLF/setup/#installation)]. The reason why we have such a tight requirement for various libraries (including python) is because we do an extensive suite of tests [[ref](https://github.com/mlcommons/GaNDLF/actions/workflows/python-test.yml)], which will be neigh impossible to run for all permutations and combinations of libraries that we depend on 😄.

Please respond to this thread or open a new one if you run into any issues! 

EDIT:
> use python 3.11

This is specifically not supported right now [[ref](https://github.com/mlcommons/GaNDLF/blob/master/setup.py#L154)].

**Reply by skoudoro** [reply]
ok, I created a new python 3.10 environment, and the installation seems to be fine now.  thanks
